zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/deployment/DbProcessState.java

```diff
@@ -336,6 +328,30 @@ public void deleteProcess(final ProcessRecord processRecord) {
         processRecord.getBpmnProcessId(), processRecord.getVersion(), processRecord.getTenantId());
   }
 
+  private void invalidateCaches(
+      final String tenantId,
+      final DirectBuffer bpmnProcessId,
+      final long processDefinitionKey,
+      final long deploymentKey,
+      final int version) {
+    final var tenantIdAndProcessIdAndVersion =
+        new TenantIdAndProcessIdAndVersion(tenantId, bpmnProcessId, version);
+    processesByTenantAndProcessIdAndVersionCache.invalidate(tenantIdAndProcessIdAndVersion);
+
+    {
+      final var key = new TenantIdAndProcessDefinitionKey(tenantId, processDefinitionKey);
+      final var cached = processByTenantAndKeyCache.getIfPresent(key);
+      if (cached != null && cached.getVersion() <= version) {
+        processByTenantAndKeyCache.invalidate(key);
```

Comment:
❓ Do we need this check? Can't we just invalidate and if it's not cached I assume the cache wouldn't invalidate anything and all will be good? We didn't have it before, and that's also how we do it with the other caches in this method.

Reply:
The check was done primarily to not invalidate the cache if the cached version is more recent than the version as argument, is it redundant?

Reply:
As-in it uses the process definition key. This unique for every definition regardless of the process id.

With forms you notice the cache is on form id and this is shared between versions.

Reply:
ah right, the `processDefinitionKey `will be different between versions :+1: